File: .claude/skills/agent-skill-kit-commons/packages/mjs/src/rules/optional_dirs.mjs
import { existsSync, readdirSync, statSync } from "node:fs";
import { join } from "node:path";
import { makeIssue } from "../types.mjs";

const OPTIONAL_DIRS = ["scripts", "references", "assets"];

export function checkOptionalDirs(skillDir) {
  const issues = [];
  for (const d of OPTIONAL_DIRS) {
    const p = join(skillDir, d);
    if (!existsSync(p)) continue;
    let st;
    try {
      st = statSync(p);
    } catch {
      continue;
    }
    if (!st.isDirectory()) continue;
    const entries = readdirSync(p).filter((n) => !n.startsWith("."));
    if (entries.length === 0) {
      issues.push(
        makeIssue(
          "W003_EMPTY_OPTIONAL_DIR",
          "warn",
          d,
          `optional directory '${d}/' exists but is empty`,
        ),
      );
    }
  }
  return issues;
}

File: .claude/skills/agent-skill-kit-commons/packages/py/src/skills_ref/rules/optional_dirs.py
from __future__ import annotations

from pathlib import Path

from ..types import Issue

_OPTIONAL_DIRS = ["scripts", "references", "assets"]


def check_optional_dirs(skill_dir: str) -> list[Issue]:
    issues: list[Issue] = []
    base = Path(skill_dir)
    for d in _OPTIONAL_DIRS:
        p = base / d
        if not p.is_dir():
            continue
        entries = [e for e in p.iterdir() if not e.name.startswith(".")]
        if not entries:
            issues.append(
                Issue(
                    "W003_EMPTY_OPTIONAL_DIR",
                    "warn",
                    d,
                    f"optional directory '{d}/' exists but is empty",
                )
            )
    return issues
